Zend/zend_inheritance.c

```diff
@@ -3418,6 +3418,23 @@ static zend_class_entry *zend_lazy_class_load(zend_class_entry *pce)
 		} while (0)
 #endif
 
+#ifdef ZEND_WIN32
+/* Hooked properties set get_iterator, which causes issues on Windows. Windows
+ * attaches multiple processes to the same shm, with each process potentially
+ * having different addresses to the corresponding get_iterator function due to
+ * ASLR. This prevents us from caching the class.
+ *
+ * To at least cache the unlinked class, avoid early-binding on Windows, and set
+ * get_iterator during inheritance. The linked class may not use inheritance
+ * cache. */
+static void zend_link_hooked_object_iter(zend_class_entry *ce, uint32_t *is_cacheable) {
+	if (!ce->get_iterator && ce->num_hooked_props) {
+		ce->get_iterator = zend_hooked_object_get_iterator;
```

Comment:
Segfault here, since ce (userland class) is in SHM. Can this even work?

Reply:
Nope, I did not look at it well enough, apparently, The call needs to be moved down, below `zend_lazy_class_load()`. I'll do that in ~1h.

Reply:
I _think_ it should be fine now, but let's see if CI passes first. :slightly_smiling_face: